timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimeLockServerConfiguration.java

```diff
@@ -139,8 +140,10 @@ private int computeNumberOfAvailableThreads() {
         Preconditions.checkState(serverFactory.getApplicationConnectors().get(0) instanceof HttpConnectorFactory,
                 "applicationConnectors of TimeLockServerConfiguration must have a HttpConnectorFactory instance.");
         HttpConnectorFactory connectorFactory = (HttpConnectorFactory) serverFactory.getApplicationConnectors().get(0);
-        int selectorThreads = connectorFactory.getSelectorThreads();
-        int acceptorThreads = connectorFactory.getAcceptorThreads();
+        int selectorThreads = connectorFactory.getSelectorThreads().orElseThrow(() -> new SafeIllegalArgumentException(
```

Comment:
I don't have much context here to determine if this was the right decision.  Would we prefer a default value instead?